applications/asset_tracker_v2/tests/cloud_codec_nrf/src/nrf_cloud_codec_test.c

```diff
@@ -0,0 +1,686 @@
+
+	ret = cloud_codec_encode_neighbor_cells(&codec, &data);
+
+	TEST_ASSERT_EQUAL(-ENOTSUP, ret);
+	TEST_ASSERT_EQUAL(NULL, codec.buf);
+	TEST_ASSERT_EQUAL(true, data.queued);
+}
+
+void test_enc_agps_req(void)
+{
+	struct cloud_data_agps_request data = {0};
+
+	ret = cloud_codec_encode_agps_request(&codec, &data);
+
+	TEST_ASSERT_EQUAL(-ENOTSUP, ret);
+	TEST_ASSERT_EQUAL(NULL, codec.buf);
+	TEST_ASSERT_EQUAL(false, data.queued);
+}
+
+void test_enc_pgps_req(void)
+{
+	struct cloud_data_pgps_request data = {0};
+
+	ret = cloud_codec_encode_pgps_request(&codec, &data);
+
+	TEST_ASSERT_EQUAL(-ENOTSUP, ret);
+	TEST_ASSERT_EQUAL(NULL, codec.buf);
+	TEST_ASSERT_EQUAL(false, data.queued);
+}
+
+void test_dec_config_null(void)
+{
+
+	ret = cloud_codec_decode_config(NULL, 0, NULL);
+	TEST_ASSERT_EQUAL(-EINVAL, ret);
+}
+
+void test_dec_config_empty(void)
+{
+	const char *str = "";
+	size_t len = 0;
+
+	ret = cloud_codec_decode_config(str, len, NULL);
+	TEST_ASSERT_EQUAL(-ENOENT, ret);
+}
+
+void test_dec_config_no_object(void)
+{
+	const char *str = "[]";
+	size_t len = strlen(str);
+
+	ret = cloud_codec_decode_config(str, len, NULL);
+	TEST_ASSERT_EQUAL(-ENOENT, ret);
+}
+
+void test_dec_config_already_handled(void)
+{
+	const char *str = "{\"version\":1}";
+	size_t len = strlen(str);
+
+	ret = cloud_codec_decode_config(str, len, NULL);
+	ret = cloud_codec_decode_config(str, len, NULL);
+	TEST_ASSERT_EQUAL(-ECANCELED, ret);
+}
+
+void test_dec_config_empty_object(void)
+{
+	const char *str = "{}";
+	size_t len = strlen(str);
+
+	ret = cloud_codec_decode_config(str, len, NULL);
+	TEST_ASSERT_EQUAL(-ENODATA, ret);
+}
+
+void test_dec_config_empty_state_object(void)
+{
+	const char *str = "{\"state\":{}}";
+	size_t len = strlen(str);
+
+	ret = cloud_codec_decode_config(str, len, NULL);
+	TEST_ASSERT_EQUAL(-ENODATA, ret);
+}
+
+void test_dec_config_data_message(void)
+{
+	const char *str = "{\"messageType\":\"\"}";
+	size_t len = strlen(str);
+
+	ret = cloud_codec_decode_config(str, len, NULL);
+	TEST_ASSERT_EQUAL(-ENOENT, ret);
+
+	str = "{\"appID\":\"\"}";
+	len = strlen(str);
+
+	ret = cloud_codec_decode_config(str, len, NULL);
+	TEST_ASSERT_EQUAL(-ENOENT, ret);
+}
+
+void test_dec_config_empty_config_object(void)
+{
+	const char *str = "{\"config\":{}}";
+	size_t len = strlen(str);
+
+	const struct cloud_data_cfg null_cfg = {0};
+	struct cloud_data_cfg cfg = null_cfg;
+
+	ret = cloud_codec_decode_config(str, len, &cfg);
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+	TEST_ASSERT_EQUAL(0, memcmp(&null_cfg, &cfg, sizeof(null_cfg)));
+}
+
+void test_dec_config_empty_state_config_object(void)
+{
+	const char *str = "{\"state\":{\"config\":{}}}";
+	size_t len = strlen(str);
+
+	const struct cloud_data_cfg null_cfg = {0};
+	struct cloud_data_cfg cfg = null_cfg;
+
+	ret = cloud_codec_decode_config(str, len, &cfg);
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+	TEST_ASSERT_EQUAL(0, memcmp(&null_cfg, &cfg, sizeof(null_cfg)));
+}
+
+void test_dec_config_good_weather(void)
```

Comment:
any reason for the name "good weather"? Do you mean "valid behavior" where everything goes well?

Reply:
Yes, this is an often used expression for tests that cover the usual case.